lib/libc/common/Kconfig

```diff
@@ -1,7 +1,13 @@
 # Copyright (c) 2023 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+config COMMON_LIBC
+	bool
+	help
+	  library of functions shared across libc variants
```

Comment:
Just to reinforce the existing pattern, can we move `config COMMON_LIBC` to `libc/Kconfig` where all the top-level libc configs exist?

Reply:
This relates to my comment below; let's resolve that before deciding where this value belongs.

Reply:
This has been removed instead.